lxd/container_put.go

```diff
@@ -51,16 +51,26 @@ func containerPut(d *Daemon, r *http.Request) Response {
 		return BadRequest(err)
 	}
 
+	if configRaw.Restore != "" && configRaw.Refresh {
+		return BadRequest(fmt.Errorf("Cannot use restore and refresh simultaneously"))
+	}
+
 	architecture, err := osarch.ArchitectureId(configRaw.Architecture)
 	if err != nil {
 		architecture = 0
 	}
 
 	var do func(*operation) error
 	var opType db.OperationType
+
 	if configRaw.Restore == "" {
 		// Update container configuration
 		do = func(op *operation) error {
+			// Never update the root device if doing refresh
+			if configRaw.Refresh {
+				configRaw.Devices["root"] = c.ExpandedDevices()["root"]
```

Comment:
So I don't think we should try to do this kind of hack. There's no guarantee that the root device will be called "root" and the exact same issue may apply to the network device or anything else that's attached to the container.

Someone who's using this feature should design their profiles in such a way that a container can be copied from one host to another without any changes.

So I'd say, let's remove the special `Refresh` attribute from ContainerPut and remove this code. If the pool of the container at the source changes, then the refresh will just fail until they reshuffle things to use profiles and avoid that situation.